sdk-wallet/keystores/encrypted_keystore.md

```diff
@@ -6,18 +6,19 @@ Note: current design is suboptimal. `EncryptedKeystore` handles both legacy keys
 
 ```
 class EncryptedKeystore:
-    // The constructor is not strictly captured by the specs; it's up to the implementing library to define it. Suggestion below.
-    // If kind == 'secretKey', `secret_key` must be provided, while `mnemonic` must be nil.
-    // If kind == 'mnemonic', `secret_key` must be nil, while `mnemonic` must be provided.
-    // In the implementation, all the parameters would be held as instance state (private fields).
-    private constructor(keys_computer: IKeysComputer, kind: string, secret_key?: ISecretKey, mnemonic?: Mnemonic)
+    // The constructor is not captured by the specs; it's up to the implementing library to define it.
```

Comment:
As stated in the last comment this can be implemented in a cleaner way by the following 2 named constructors: one for the mnemonic and one for the secret key. This is clearly not required